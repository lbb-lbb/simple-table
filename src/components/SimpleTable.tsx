import { defineComponent, ref } from "vue";
import { type TableProps, tableProps } from "./types";
import TableBody from './table_body'
import TableHeader from './table_header'
import TableFooter from './table_footer'
export default defineComponent({
  name: "SimpleTable",
  props: tableProps,
  components: {
    TableBody,
    TableFooter,
    TableHeader
  },
  setup(props: TableProps, { attrs, emit, slots }) {
    const a = ref('æˆ‘æ˜¯å¤´éƒ¨')
    return () => {
      return (
        <div>
          <table border={1} width={600} height={400}>
            <table-header a={a} />
            <table-body />
          </table>
          <table-footer />
        </div>
      );
    };
  },
});
