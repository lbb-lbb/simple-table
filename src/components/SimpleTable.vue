<template>
  <div>
    <table class="table">
      <table-header v-bind="$attrs"
                    :columns="columns"
                    @changeSort="changeSort">
        <template v-for="item in columns"
                  :key="item.value"
                  v-slot:[addHeaderSlotName(item.value)]="scope">
          <slot :name="addHeaderSlotName(item.value)"
                :item="scope.item">
          </slot>
        </template>
      </table-header>
      <table-body v-bind="$attrs"
                  :order-by="orderBy"
                  :order="order"
                  :columns="columns">
        <template v-for="item in columns"
                  :key="item.value"
                  v-slot:[item.value]="scope">
          <slot :name="item.value"
                :item="scope.item">
          </slot>
        </template>
        <template v-slot:options="scope">
          <slot name="options"
                :item="scope.item">
          </slot>
        </template>
      </table-body>
    </table>
    <table-footer v-bind="$attrs"/>
  </div>
</template>

<script lang="ts" setup>
/**
 * @file 表格组件
 */
import {ref} from "vue";
import { tableProps, ColumnsType} from "./types";
import TableBody from './table_body/index.vue'
import TableHeader from './table_header/index.vue'
import TableFooter from './table_footer/index.vue'

  const props = defineProps({
    ...tableProps
  })
  const orderBy = ref('')
  const order = ref('sort-normal')

  function addHeaderSlotName(value: string) {
    return `header-${value}`
  }
  /**
   * 接收header传递出来的排序字段和方式并改变，触发body里面的排序计算
   * @param item 排序的表头数据
   * @param type 排序的方式标识
   */
  function changeSort(item: ColumnsType, type: string) {
    orderBy.value = item.value
    order.value = type
  }

</script>
<style>
table, table tr th, table tr td {
  border: 1px solid #ccc;
}

</style>