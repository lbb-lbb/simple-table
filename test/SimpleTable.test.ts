import { shallowMount } from "@vue/test-utils";
import { expect, test } from "vitest";
import SimpleTable from "../src/components/lxy_table/lxy_table.vue";

test("mount component", () => {
  const wrapper = shallowMount(SimpleTable, {
    props: {},
  });

  expect(wrapper.html()).toMatchSnapshot();
});
