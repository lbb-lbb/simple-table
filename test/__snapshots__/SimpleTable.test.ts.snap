// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div data=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\">
  <table class=\\"table\\">
    <table-header-stub columns=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" openoption=\\"false\\" data=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\"></table-header-stub>
    <table-body-stub data=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" columns=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" orderby=\\"\\" order=\\"sort-normal\\" openoption=\\"false\\"></table-body-stub>
  </table>
</div>"
`;
